### PR TITLE
add logical consistency prompt

### PR DESCRIPTION
**Pull request by erees1**

### CODE REVIEW

File: stage_one.py [hunk 1]
```diff
@@ -137,7 +133,13 @@ def main(
     repeats_per_question: int = 1,
     temperature: Optional[float] = None,
 ):
-    tasks = validate_dataset_and_task(dataset, tasks)
+    if dataset is not None:
+        assert tasks is None
```
Comment (thejaminator) [inline stage_one.py:137]: add comment here

File: cot_transparency/model_apis.py [hunk 2]
```diff
@@ -22,36 +22,40 @@ def messages_has_none_role(prompt: list[StrictChatMessage] | list[ChatMessage])
...
+        if all_strict:
+            strict_prompt: list[StrictChatMessage] = prompt  # type: ignore
+            return strict_prompt
+        else:
+            flex_prompt: list[ChatMessage] = prompt  # type: ignore
+
+            if model == "gpt-3.5-turbo" or model == "gpt-4" or model == "gpt-3.5-turbo-16k":
+                strict_prompt = format_for_openai_chat(flex_prompt)
+            elif model == "claude-v1" or model in [
+                "text-davinci-003",
```
Comment (thejaminator) [inline cot_transparency/model_apis.py:40]: add claude-2

File: cot_transparency/model_apis.py [hunk 3]
```diff
@@ -22,36 +22,40 @@ def messages_has_none_role(prompt: list[StrictChatMessage] | list[ChatMessage])
...
+                strict_prompt = format_for_completion(flex_prompt)
+            else:
+                raise ValueError(f"Unknown model {model}")
+        return strict_prompt
 
     def convert_to_anthropic_str(self) -> str:
         if messages_has_none_role(self.messages):
             raise ValueError(f"Anthropic chat messages cannot have a None role. Got {self.messages}")
-        return self.convert_to_completion_str()
+        return self.convert_to_completion_str(model="claude-v1")
```
Comment (thejaminator) [inline cot_transparency/model_apis.py:53]: here too

Comment (erees1) [reply cot_transparency/model_apis.py]: added the ModelType enum

**Review by thejaminator — APPROVED** [review]

File: cot_transparency/model_apis.py (hunk 3 above)
Comment (erees1) [reply cot_transparency/model_apis.py]: done

**Pull request merged**